File: 2021Jun21.hs
-- Given a list of numbers, find if there exists a pythagorean triplet in that list. A pythagorean triplet is 3 variables a, b, c where a2 + b2 = c2
--
-- Example:
-- Input: [3, 5, 12, 5, 13]
-- Output: True
-- Here, 5^2 + 12^2 = 13^2
--

findPythagoreanTriplets :: (Num a, Eq a) => [a] -> Bool
findPythagoreanTriplets x = 
    let sqList = map (^2) x
        sqPairs x = case x of
            []       -> []
            (x:y:[]) -> x^2 + y^2:sqPairs []
            (x:y:xs) -> x^2 + y^2:sqPairs (y:xs)
    in True `elem` [sl `elem` (sqPairs x) | sl <- sqList]

File: 2021/2021Jun21.py
#!/usr/bin/env python3
##
## Given a list of numbers, find if there exists a pythagorean triplet in that list. A pythagorean triplet is 3 variables a, b, c where a2 + b2 = c2
##
## Example:
## Input: [3, 5, 12, 5, 13]
## Output: True
## Here, 5^2 + 12^2 = 13^2
##

def findPythagoreanTriplets(nums):
    # Fill this in.
    pyt = [(nums[i], nums[i+1], nums[i+2]) for i in range(len(nums)-2) if (nums[i]**2+nums[i+1]**2 == nums[i+2]**2) or (nums[i+1]**2+nums[i+2]**2 == nums[i]**2) or (nums[i+2]**2+nums[i]**2 == nums[i+1]**2)]
    return (len(pyt) != 0)


print(findPythagoreanTriplets([3,6,12,5,13]))
# True

